lib/archethic/db/embedded_impl/inputs_reader.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Archethic.DB.EmbeddedImpl.InputsReader do
+  @moduledoc """
+  Inputs are stored by destination address. 1 file per address per ledger
+  """
+
+  alias Archethic.DB.EmbeddedImpl.InputsWriter
+  alias Archethic.TransactionChain.VersionedTransactionInput
+  alias Archethic.Utils
+
+  @spec get_inputs(ledger :: InputsWriter.ledger(), address :: binary()) ::
+          list(VersionedTransactionInput.t())
+  def get_inputs(ledger, address) do
+    filename = InputsWriter.address_to_filename(ledger, address)
+
+    case File.open(filename, [:read, :binary]) do
+      {:error, :enoent} ->
+        []
+
+      {:ok, fd} ->
+        case IO.binread(fd, :eof) do
+          :eof ->
+            []
+
+          bin ->
+            bin
+            |> deserialize_inputs_file([])
+            |> Enum.reverse()
+        end
+    end
+  end
+
+  defp deserialize_inputs_file(<<>>, acc), do: acc
+
+  defp deserialize_inputs_file(bitstring, acc) do
+    {input_bit_size, rest} = Utils.VarInt.get_value(bitstring)
+
+    # every serialization contains some padding to be a binary (multipe of 8bits)
+    {input_bitstring, rest} =
+      case rem(input_bit_size, 8) do
```

Comment:
Here you need to do some calculation because the `input_bit_size` is the size of the input itself without the wrapped bits.
Maybe in the inputs writer you should wrap the input before getting it's size, as it you don't need to use `rem` to determine the wrapped bits size. The function `VersionedTransactionInput.deserialize(input_bitstring)` may returns a non empty rest which will correspond to the wrapped bit that can be ignored